From e094806951ff7337b5b0c534db479e3808f153a7 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Tue, 13 Nov 2018 16:00:23 +0100
Subject: [PATCH] Fix compile issue with gcc 9

GCC 9's libstdc++ requires a seed sequence provider to expose
result_type. This is the upstream Qt change retained locally because it
was committed after the Qt 5.11.3 source archive was released.

Change-Id: If38151d1fa6f40a80274acc26d9ed6b4ac6049fe
---
 src/corelib/global/qrandom.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/corelib/global/qrandom.cpp b/src/corelib/global/qrandom.cpp
index ebf9864b15..577736a867 100644
--- a/src/corelib/global/qrandom.cpp
+++ b/src/corelib/global/qrandom.cpp
@@ -218,6 +218,7 @@ struct QRandomGenerator::SystemGenerator
 #endif // Q_OS_WINRT
 
     static SystemGenerator &self();
+    typedef quint32 result_type;
     void generate(quint32 *begin, quint32 *end) Q_DECL_NOEXCEPT_EXPR(FillBufferNoexcept);
 
     // For std::mersenne_twister_engine implementations that use something
-- 
2.19.1

